Amazon.Ion.ObjectMapper/IonListSerializer.cs

```diff
@@ -1,35 +1,45 @@
 using System;
+using System.Collections;
 using System.Collections.Generic;
 using Amazon.IonDotnet;
 
 namespace Amazon.Ion.ObjectMapper
 {
-    public class IonListSerializer : IonSerializer<System.Collections.IList>
+    public class IonListSerializer : IonSerializer<IList>
     {
         private readonly IonSerializer serializer;
         private readonly Type listType;
         private readonly Type elementType;
         private readonly bool isGenericList;
 
-        public IonListSerializer(IonSerializer serializer, Type listType, Type elementType)
-        {
-            this.serializer = serializer;
-            this.listType = listType;
-            this.elementType = elementType;
-            this.isGenericList = true;
-        }
-
         public IonListSerializer(IonSerializer serializer, Type listType)
         {
             this.serializer = serializer;
             this.listType = listType;
-            this.isGenericList = false;
+
+            if (this.listType.IsArray)
+            {
+                this.elementType = this.listType.GetElementType();
+                this.isGenericList = true;
+            }
+            else if (this.listType.IsAssignableTo(typeof(IList)))
+            {
+                this.isGenericList = this.listType.IsGenericType;
+                if (this.isGenericList)
+                {
+                    this.elementType = this.listType.GetGenericArguments()[0];
+                }
+            }
+            else
+            {
+                throw new NotSupportedException("Encountered an Ion list but the desired deserialized type was not an IList, it was: " + listType);
+            }
         }
 
-        public System.Collections.IList Deserialize(IIonReader reader)
+        public IList Deserialize(IIonReader reader)
```

Comment:
Can you revert these changes as they're unrelated to the PR and is not in code that you've changed, so it adds noise.